zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +148,205 @@ def compute(self, today, assets, out, close, volume):
         out[:] = nanmean(close * volume, axis=0)
 
 
+class CorrelationFactor(SingleInputMixin, CustomFactor):
```

Comment:
> In terms of issues with normality, yes but that's something that a user is just going to have to learn through the lectures imo.

If it's wrong to use Pearson in our domain, I don't think we should be making it easy to do an operation that's subtly wrong without thinking about it.  If we're giving people footguns, they should at least come with warning labels.

> my only worry is that might be confusing for beginner users, adding to the complexity of pipeline. People are used to thinking about 'correlation', not the underlying type.

Again, "the user doesn't know what they're doing" is not an argument for making it easy to make mistakes.  Giving things explicit names forces the user to make explicit choices, which sometimes forces them to learn about what the choices are.  We should optimize for the user's ability to make something they understand, not for their ability to make something that doesn't crash.